src/components/Layout/NestedScrollLayoutContext.tsx

```diff
@@ -0,0 +1,41 @@
+import { createContext, PropsWithChildren, useEffect, useMemo, useRef } from "react";
+import { Css, Only, Xss } from "src/Css";
+
+interface NestedScrollLayoutContextProps {
+  scrollableEl: HTMLElement;
+}
+
+export const NestedScrollLayoutContext = createContext<NestedScrollLayoutContextProps>({
+  scrollableEl: document.createElement("div"),
+});
+
+// Allow any css to be applied to the NestedScrollLayout container.
+type NestedScrollXss = Xss<any>;
+interface NestedScrollLayoutContextProviderProps<X> {
+  xss?: X;
+}
+
+export function NestedScrollProvider<X extends Only<NestedScrollXss, X>>({
```

Comment:
I think it'd be nice to hide the `Provider` / "this uses `content`" aspect of this name...

Maybe just `ScrollableParent`? To match the `ScorllableContent`?